tests/test_checkpointing.py

```diff
@@ -0,0 +1,126 @@
+                                   service_name=utils.SPARK_SERVICE_NAME,
+                                   args=producer_args)
+
+    # validating producer output
+    utils.check_job_output(producer_id, "{} messages sent to Kafka".format(len(messages)))
+
+
+@pytest.mark.skipif(not utils.hdfs_enabled(), reason='HDFS_ENABLED is false')
+@pytest.mark.sanity
+def test_structured_streaming_recovery(kerberized_spark, kerberized_kafka):
+
+    @retrying.retry(
+        wait_fixed=5000,
+        stop_max_delay=600 * 1000,
+        retry_on_result=lambda res: not res)
+    def wait_for_running_job_output(task_id, expected_line):
+        stdout = sdk_cmd.run_cli("task log --lines=1000 {}".format(task_id))
+        result = expected_line in stdout
+        LOGGER.info('Checking for {} in STDOUT:\n{}\nResult: {}'.format(expected_line, stdout, result))
+        return result
+
+    kafka_brokers = ','.join(sdk_cmd.svc_cli(KAFKA_PACKAGE_NAME, KAFKA_SERVICE_NAME, 'endpoints broker', json=True)['dns'])
+    LOGGER.info("Kafka brokers: {}".format(kafka_brokers))
+
+    _uri = upload_jaas()
+    uris = "spark.mesos.uris={}".format(_uri)
+
+    jar_uri = utils.upload_dcos_test_jar()
+
+    kafka_kerberos_args = get_kerberized_kafka_spark_conf(utils.SPARK_SERVICE_NAME)
+    LOGGER.info("Spark Kerberos configuration for Kafka:\n{}".format('\n'.join(kafka_kerberos_args)))
+
+    common_args = [
+        "--conf", "spark.mesos.containerizer=mesos",
+        "--conf", "spark.scheduler.maxRegisteredResourcesWaitingTime=2400s",
+        "--conf", "spark.scheduler.minRegisteredResourcesRatio=1.0",
+        "--conf", uris
+    ] + kafka_kerberos_args
+
+    # configuring streaming job and HDFS folders
+    setup_hdfs_paths()
+
+    # running kafka producer
+    message_set_a = ["abc"] * 100
+    feed_sample_data(jar_uri, kafka_brokers, KAFKA_TEST_TOPIC, common_args, message_set_a)
+
+    spark_submit_args = ["--supervise",
+                         "--class", "StructuredStreamingWithCheckpointing",
+                         "--conf", "spark.cores.max=2",
+                         "--conf", "spark.executor.cores=1",
+                         "--conf", "spark.sql.shuffle.partitions=2",
+                         "--conf", "spark.executor.memory=2g"] + common_args
+
+    application_args = "{} {} {} {}".format(kafka_brokers, KAFKA_TEST_TOPIC, HDFS_CHECKPOINT_DIR, SPARK_SECURITY_PROTOCOL)
+
+    driver_task_id = utils.submit_job(app_url=jar_uri,
+                                      app_args=application_args,
+                                      service_name=utils.SPARK_SERVICE_NAME,
+                                      args=(SPARK_SUBMIT_HDFS_KERBEROS_ARGS + spark_submit_args))
+
+    # Wait until executor is running
+    LOGGER.info("Starting supervised driver {}".format(driver_task_id))
+    sdk_tasks.check_running(SPARK_APPLICATION_NAME, 1, timeout_seconds=600)
```

Comment:
can you add `expected_task_count=1` otherwise the lone `1` is a bit mysterious or use a named_constant because I see something similar on L117.

Reply:
good catch. fixed that.